src/borg/testsuite/archiver/transfer_cmd.py

```diff
@@ -85,10 +85,10 @@ def local_to_utc(local_naive, tzoffset, tzinfo):
             del expected_archive["barchive"]
             # timestamps:
             # borg 1.2 transformed to local time and had microseconds = 0, no tzoffset
-            # borg 2 uses an utc timestamp, with microseconds and with tzoffset
+            # borg 2 uses local time, with microseconds and with tzoffset
             for key in "start", "time":
-                # fix expectation: local time meant +01:00, so we convert that to utc +00:00
-                expected_archive[key] = local_to_utc(expected_archive[key], repo12_tzoffset, timezone.utc)
+                # fix expectation: local time meant +01:00, so we convert that to whatever local tz is here.
```

Comment:
I don't understand this comment.  What does "fix expectation" mean?  And why does it say "local time meant +01:00"?

Reply:
the json test data i generated back then on my machine with borg 1.2 was in +01:00 timezone (germany, non-DST) - is is contained in a tar file in master branch. borg 1.2 did not include the tzoffset in its json timestamps, so that is something one needs to just know.

then there is what we **got** (from generated borg2 json output) and what we **expected** (from the borg 1.2 json file). as both jsons were generated from the same repository (in case of borg2, it is after "borg transfer" of the archives), both jsons represent the same data, but some differences are expected in the json due to the breaking changes in borg2.

at the end of this test, i basically do `assert got == expected` for the whole deserialized json dict. for that to work, i transform the expected value to be able to precisely match the value we got from borg2 (for timestamps, this is basically: asume +01:00 for the naive ts we got, then transform to a non-naive ts with local tzoffset of the test machine [which is currently also +01:00 on my machine, but could be any other offset]).